pkg/core/function.go

```diff
@@ -185,21 +185,27 @@ func (c *client) CreateFunction(buildpackBuilder Builder, options CreateFunction
 
 		if options.Verbose || options.Wait {
 			stopChan := make(chan struct{})
-			errChan := make(chan error)
+			loggingTerminationChan := make(chan error)
 			if options.Verbose {
-				go c.displayFunctionCreationProgress(ns, s.Name, log, stopChan, errChan)
+				go c.displayFunctionCreationProgress(ns, s.Name, log, stopChan, loggingTerminationChan)
+			}
+			err := c.waitForSuccessOrFailure(ns, s.Name, 1, stopChan, loggingTerminationChan, options.Verbose)
+			if options.Verbose {
+				// wait until log printing has finished before returning as this will terminate the process
+				if logErr := <-loggingTerminationChan; logErr != nil && err == nil {
+					return nil, nil, nil, logErr
+				}
 			}
-			err := c.waitForSuccessOrFailure(ns, s.Name, 1, stopChan, errChan, options.Verbose)
 			if err != nil {
 				return nil, nil, nil, err
 			}
 
 			// fetch latest Service and Revision, ignore errors
-			s1, err := c.serving.ServingV1alpha1().Services(s.Namespace).Get(s.Name, metav1.GetOptions{})
+			s1, err := c.serving.ServingV1alpha1().Services(s.Namespace).Get(s.Name, v1.GetOptions{})
```

Comment:
please revert to `metav1`, the name `v1` is too generic and leads to conflicts with other `v1` packages.

Reply:
Fixed.